Jellyfin.Plugin.OpenSubtitles/OpenSubtitleDownloader.cs

```diff
@@ -207,124 +168,143 @@ await Task.Delay(250, cancellationToken)
+                {
+                    case HttpStatusCode.NotAcceptable when info.Data.Remaining <= 0:
+                    {
+                        if (_login?.User != null)
+                        {
+                            _login.User.RemainingDownloads = 0;
+                        }
 
-            if (_rateLimitLeft == 0)
-            {
-                await Task.Delay(1000, cancellationToken)
-                    .ConfigureAwait(false);
-            }
+                        throw new OpenApiException("OpenSubtitles download limit reached");
+                    }
 
-            var downloadResponse = await OpenSubtitlesHandler.OpenSubtitles.DownloadSubtitlesAsync(downloadsList, cancellationToken).ConfigureAwait(false);
+                    case HttpStatusCode.Unauthorized:
+                        // JWT token expired, obtain a new one and try again?
+                        _login = null;
+                        return await GetSubtitlesInternal(id, cancellationToken).ConfigureAwait(false);
+                }
 
-            var downloadResult = downloadResponse.Item1;
-            _rateLimitLeft = downloadResponse.Item2 ?? _rateLimitLeft;
+                var msg = info.Body.Contains("<html", StringComparison.OrdinalIgnoreCase) ? "[html]" : info.Body;
 
-            if (downloadResponse.Item2 != null)
-            {
-                if (_rateLimitLeft <= 4)
-                {
-                    await Task.Delay(250, cancellationToken)
-                        .ConfigureAwait(false);
-                }
+                msg = string.Format(
+                    CultureInfo.InvariantCulture,
+                    "Invalid response for file {0}: {1}\n\n{2}",
+                    fid,
+                    info.Code,
+                    msg);
+
+                throw new OpenApiException(msg);
             }
 
-            if ((downloadResult.Status ?? string.Empty).IndexOf("407", StringComparison.OrdinalIgnoreCase) != -1)
+            if (_login?.User != null)
             {
-                _lastRateLimitException = DateTime.UtcNow;
-                throw new RateLimitExceededException("OpenSubtitles daily limit reached");
+                _login.User.RemainingDownloads = info.Data.Remaining;
+                _logger.LogInformation($"Remaining downloads: {_login.User.RemainingDownloads}");
             }
 
-            if (downloadResult is not MethodResponseSubtitleDownload subtitleDownloadResult)
+            if (string.IsNullOrWhiteSpace(info.Data.Link))
             {
-                throw new OpenApiException($"Invalid response type. Name: {downloadResult.Name}, Message: {downloadResult.Message}, Status: {downloadResult.Status}");
+                var msg = string.Format(
+                    CultureInfo.InvariantCulture,
+                    "Failed to obtain download link for file {0}: {1}",
+                    fid,
+                    info.Code);
+
+                throw new OpenApiException(msg);
```

Comment:
I don't think OpenApiException is appropriate anywhere. It's an exception that is used in the OpenApi library.

Reply:
Replaced and removed in 8d19bbb5f4f5fbe649baee2e890b288109b07d39 and 260575a7d38e98a07c871e15d3f149e2cc2a5ac6